### PR TITLE
add_quantiles

### PR DESCRIPTION
**Pull request by timtreis**
as discussed in #93 

tagging @joanacmbarros since I can't tag her as a reviewer?

### CODE REVIEW

File: R/add_quantiles.R
Comment (SHAESEN2) [inline R/add_quantiles.R:1]: Make more general as you are describing a method here.

File: R/add_quantiles.R
Comment (SHAESEN2) [inline R/add_quantiles.R:3]: for any plot right?

File: R/add_quantiles.R
Comment (SHAESEN2) [inline R/add_quantiles.R:38]: Vector of quantiles to be displayed on the probability scale

File: R/add_quantiles.R
Comment (SHAESEN2) [inline R/add_quantiles.R:51]: if i want black, do i put "black"? this is not clear. Perhaps you can specify that this needs to be a character representing ....

File: R/add_quantiles.R
Comment (SHAESEN2) [inline R/add_quantiles.R:55]: check if entire vector is numeric: base::all

File: R/add_quantiles.R
Comment (SHAESEN2) [inline R/add_quantiles.R:69]: we will get errors if only strata if we dont add the overall strata manually see add_risktable

File: R/add_quantiles.R
Comment (SHAESEN2) [inline R/add_quantiles.R:83]: this will depend on the scale of the graph. Perhaps you can pull out the Y axis range. You need to check if it is on the probability scale, cumulative hazard etc. Is this info stored in the graph?

File: tests/testthat/test-add_quantiles.R
Comment (SHAESEN2) [inline tests/testthat/test-add_quantiles.R:1]: update

File: tests/testthat/test-add_quantiles.R
Comment (SHAESEN2) [inline tests/testthat/test-add_quantiles.R:13]: should be more elaborate

**Review by SHAESEN2 — CHANGES_REQUESTED** [review]

File: R/add_quantiles.R
Comment (timtreis) [reply R/add_quantiles.R:83]: fixed, good catch

File: R/add_quantiles.R
Comment (SHAESEN2) [inline R/add_quantiles.R:40]: default: strata , <- remove comma

File: R/add_quantiles.R
Comment (SHAESEN2) [reply R/add_quantiles.R:83]: I dont see the fix

**Review by SHAESEN2 — CHANGES_REQUESTED** [review]

File: tests/testthat/test-add_quantiles.R
Comment (SHAESEN2) [inline tests/testthat/test-add_quantiles.R:13]: Should we make sure all tests are in place before merging?

**Review by SHAESEN2 — COMMENTED** [review]
Fine for me but make sure that all testing is in place before merging.

File: R/add_quantiles.R
Comment (timtreis) [reply R/add_quantiles.R]: Now it is

File: tests/testthat/test-add_quantiles.R
Comment (timtreis) [reply tests/testthat/test-add_quantiles.R:13]: Steven, you're so impatient :D Yeah, I'll pring you when all possible use cases are functioning and test coverage is 100 %

File: R/add_quantiles.R
Comment (SHAESEN2) [reply R/add_quantiles.R:3]: only for survival? then why create a method?

File: R/visr.R
Comment (SHAESEN2) [inline R/visr.R:171]: keep base::switch as discussed. We never know if switch will occur in other packages.

File: R/visr.R
Comment (SHAESEN2) [inline R/visr.R:187]: keep base::switch

File: R/visr.R
Comment (SHAESEN2) [inline R/visr.R:239]: Here OK to remove base

File: R/visr.R
Comment (SHAESEN2) [inline R/visr.R:297]: it needs to be rounded to avoid clutter on the y axis. I agree that pretty is not the best choice. Did you find alternatives?
You could determine the amount of digits and then limit them to amount-1

File: R/visr.R
Comment (SHAESEN2) [inline R/visr.R:323]: will we have this as well with other primitive functions? Imagine I use log(-log) which is a common transformation in survival estimation?

File: tests/testthat/test-add_quantiles.R
Comment (SHAESEN2) [inline tests/testthat/test-add_quantiles.R:20]: A test for log or other primitive functions

**Review by SHAESEN2 — CHANGES_REQUESTED** [review]

File: R/add_quantiles.R
Comment (timtreis) [reply R/add_quantiles.R:3]: Right now we only have survival implemented, but maybe we'll add others? So I stuck with the structure. Also, I copied the file from `add_CI` and then just adjusted from there on. Theoretically we're not yet using that for anything else yet, but who knows.

File: R/visr.R
Comment (timtreis) [reply R/visr.R:297]: In the end, `pretty` wasn't really the problem since that respects the value range. But, given that we were rounding with `0` precision, this would for example round 1.499999 to 1. 

The alternative approach would be a significantly more elaborate helper function that 1) rounds, 2) pretties and 3) then extend the upper/lower bound my extrapolating with the `pretty` result spacing between numbers to potentially add a 6th/7th before and/or after.

File: R/visr.R
Comment (timtreis) [reply R/visr.R:323]: True, that currently breaks. I'll fix that

Comment (timtreis) [reply R/visr.R]: fixed it

**Comment by timtreis** [comment]
Check is at 0/0/0, codecov at 100 %

File: R/add_quantiles.R
Comment (SHAESEN2) [reply R/add_quantiles.R:3]: A method should be general so please describe it generally

Comment (SHAESEN2) [reply R/add_quantiles.R:3]: update to keep general.

File: R/visr.R
Comment (SHAESEN2) [inline R/visr.R:320]: This is really cool and we should highlight this in the documentation as well. Survminer is taking the lazy route:

  if(is.null(fun)) draw_lines <- TRUE
  else if(fun %in% c("cumhaz", "cloglog")){
    warning("Adding survival median lines is not allowed when fun is: ", fun)
    draw_lines <- FALSE
  }
  else if(fun == "pct") med_y <- 50

File: tests/testthat/test-add_quantiles.R
Comment (SHAESEN2) [inline tests/testthat/test-add_quantiles.R:20]: This seems to be a repeat of earlier tests. I like this general T3. and then putting everything below it regarding parameter choices.

Use argument instead of parameter

**Review by SHAESEN2 — CHANGES_REQUESTED** [review]

**Comment by SHAESEN2** [comment]
Nice work. Some details pending :)

**Comment by SHAESEN2** [comment]
Hey @timtreis I have reviewed the branch. For me OK to merge now unless you have comments on my commit.

**Pull request merged**
